`#[message_handler]` hygiene: macro assumes import of  `futuresdr::anyhow::Result`

This code fails to compile for reasons that are difficult to understand at first:

```rust
use futuresdr::macros::message_handler;
use futuresdr::runtime::{Block, BlockMetaBuilder, BlockMeta, Kernel, MessageIo, MessageIoBuilder, Pmt, StreamIoBuilder, WorkIo};

struct MsgThing;

impl MsgThing {
    fn new() -> Block {
        Block::new(
            BlockMetaBuilder::new("MsgThing").build(),
            StreamIoBuilder::new()
                .build(),
            MessageIoBuilder::new()
                .add_input("in", Self::in_port)
                .build(),
            Self
        )
    }

    #[message_handler]
    async fn in_port(
        &mut self,
        _io: &mut WorkIo,
        _mio: &mut MessageIo<Self>,
        _meta: &mut BlockMeta,
        p: Pmt,
    ) -> futuresdr::anyhow::Result<Pmt> {
        Ok(Pmt::Ok)
    }
}

impl Kernel for MsgThing {}
```

Error message:

```rust
error[E0107]: enum takes 2 generic arguments but 1 generic argument was supplied
   --> src/main.rs:19:5
    |
19  |     #[message_handler]
    |     ^^^^^^^^^^^^^^^^^^
    |     |
    |     expected 2 generic arguments
    |     supplied 1 generic argument
    |
note: enum defined here, with 2 generic parameters: `T`, `E`
   --> /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/result.rs:527:10
    |
527 | pub enum Result<T, E> {
    |          ^^^^^^ -  -
    = note: this error originates in the attribute macro `message_handler` (in Nightly builds, run with -Z macro-backtrace for more info)
help: add missing generic argument
    |
19  |     #[message_handler], E
    |                       +++

```

If you add `use futuresdr::anyhow::Result` to the current scope it compiles, but this conflicts with the use of crate-local `Result` types, and is confusing due to the return type of `in_port` being fully qualified. 

I suspect the problem is that the macro doesn't fully qualify the `Result` type here:

https://github.com/FutureSDR/FutureSDR/blob/f44b52c5e3c979af239b7706d1572383ec2efe90/crates/macros/src/lib.rs#L552

--- a/crates/macros/src/lib.rs
+++ b/crates/macros/src/lib.rs
@@ -549,7 +549,7 @@ pub fn message_handler(
             #mio: &'a mut MessageIo<Self>,
             #meta: &'a mut BlockMeta,
             #pmt: Pmt,
-        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = Result<Pmt>> + Send + 'a>> {
+        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = crate::anyhow::Result<Pmt>> + Send + 'a>> {
             use crate::futures::FutureExt;
             Box::pin(async move {
                 #(#body)*
@@ -562,7 +562,7 @@ pub fn message_handler(
             #mio: &'a mut MessageIo<Self>,
             #meta: &'a mut BlockMeta,
             #pmt: Pmt,
-        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = Result<Pmt>> + 'a>> {
+        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = crate::anyhow::Result<Pmt>> + 'a>> {
             use crate::futures::FutureExt;
             Box::pin(async move {
                 #(#body)*
@@ -639,7 +639,7 @@ pub fn message_handler_external(
             #mio: &'a mut MessageIo<Self>,
             #meta: &'a mut BlockMeta,
             #pmt: Pmt,
-        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = Result<Pmt>> + Send + 'a>> {
+        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = futuresdr::anyhow::Result<Pmt>> + Send + 'a>> {
             use futuresdr::futures::FutureExt;
             Box::pin(async move {
                 #(#body)*
@@ -652,7 +652,7 @@ pub fn message_handler_external(
             #mio: &'a mut MessageIo<Self>,
             #meta: &'a mut BlockMeta,
             #pmt: Pmt,
-        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = Result<Pmt>> + 'a>> {
+        ) -> std::pin::Pin<Box<dyn std::future::Future<Output = futuresdr::anyhow::Result<Pmt>> + 'a>> {
             use futuresdr::futures::FutureExt;
             Box::pin(async move {
                 #(#body)*

--- a/src/blocks/message_annotator.rs
+++ b/src/blocks/message_annotator.rs
@@ -1,6 +1,5 @@
 use std::collections::HashMap;
 
-use crate::anyhow::Result;
 use crate::runtime::Block;
 use crate::runtime::BlockMeta;
 use crate::runtime::BlockMetaBuilder;

--- a/src/blocks/message_copy.rs
+++ b/src/blocks/message_copy.rs
@@ -1,4 +1,3 @@
-use crate::anyhow::Result;
 use crate::runtime::Block;
 use crate::runtime::BlockMeta;
 use crate::runtime::BlockMetaBuilder;

--- a/src/blocks/message_pipe.rs
+++ b/src/blocks/message_pipe.rs
@@ -1,7 +1,6 @@
 use futures::channel::mpsc;
 use futures::SinkExt;
 
-use crate::anyhow::Result;
 use crate::runtime::Block;
 use crate::runtime::BlockMeta;
 use crate::runtime::BlockMetaBuilder;

--- a/tests/message_handler.rs
+++ b/tests/message_handler.rs
@@ -5,7 +5,10 @@ mod isolated_scope {
     #[test]
     fn message_handler_compiles() {
         use futuresdr::macros::message_handler;
-        use futuresdr::runtime::{Block, BlockMetaBuilder, BlockMeta, Kernel, MessageIo, MessageIoBuilder, Pmt, StreamIoBuilder, WorkIo};
+        use futuresdr::runtime::{
+            Block, BlockMeta, BlockMetaBuilder, Kernel, MessageIo, MessageIoBuilder, Pmt,
+            StreamIoBuilder, WorkIo,
+        };
 
         struct MsgThing;
 
@@ -13,12 +16,11 @@ mod isolated_scope {
             fn new() -> Block {
                 Block::new(
                     BlockMetaBuilder::new("MsgThing").build(),
-                    StreamIoBuilder::new()
-                        .build(),
+                    StreamIoBuilder::new().build(),
                     MessageIoBuilder::new()
                         .add_input("in", Self::in_port)
                         .build(),
-                    Self
+                    Self,
                 )
             }
 

--- a/tests/message_handler.rs
+++ b/tests/message_handler.rs
@@ -1,0 +1,43 @@
+/// This module exists to try to inhibit accidental import of [`futuresdr::anyhow::Result`]
+#[cfg(test)]
+mod isolated_scope {
+    /// Regression test for https://github.com/FutureSDR/FutureSDR/issues/149
+    #[test]
+    fn message_handler_compiles() {
+        use futuresdr::macros::message_handler;
+        use futuresdr::runtime::{Block, BlockMetaBuilder, BlockMeta, Kernel, MessageIo, MessageIoBuilder, Pmt, StreamIoBuilder, WorkIo};
+
+        struct MsgThing;
+
+        impl MsgThing {
+            fn new() -> Block {
+                Block::new(
+                    BlockMetaBuilder::new("MsgThing").build(),
+                    StreamIoBuilder::new()
+                        .build(),
+                    MessageIoBuilder::new()
+                        .add_input("in", Self::in_port)
+                        .build(),
+                    Self
+                )
+            }
+
+            #[message_handler]
+            async fn in_port(
+                &mut self,
+                _io: &mut WorkIo,
+                _mio: &mut MessageIo<Self>,
+                _meta: &mut BlockMeta,
+                _p: Pmt,
+            ) -> Result<Pmt> {
+                Ok(Pmt::Ok)
+            }
+        }
+
+        impl Kernel for MsgThing {}
+
+        // Main test is that the above compiles without futuresdr::anyhow::Result being imported in scope.
+        let b = MsgThing::new();
+        assert!(b.message_input_name_to_id("in").is_some());
+    }
+}

